azure/services/async/async_test.go

```diff
@@ -117,7 +117,7 @@ func TestServiceCreateOrUpdateResource(t *testing.T) {
 					s.GetLongRunningOperationState(resourceName, serviceName, infrav1.PutFuture).Return(nil),
 					c.Get(gomockinternal.AContext(), gomock.AssignableToTypeOf(azureResourceGetterType)).Return(nil, &azcore.ResponseError{StatusCode: http.StatusNotFound}),
 					r.Parameters(gomockinternal.AContext(), nil).Return(fakeParameters, nil),
-					c.CreateOrUpdateAsync(gomockinternal.AContext(), gomock.AssignableToTypeOf(azureResourceGetterType), "", gomock.Any()).Return(nil, fakePoller[MockCreator](g, http.StatusAccepted), context.DeadlineExceeded),
+					c.CreateOrUpdateAsync(gomockinternal.AContext(), gomock.AssignableToTypeOf(azureResourceGetterType), gomock.Any()).Return(nil, fakePoller[MockCreator](g, http.StatusAccepted), context.DeadlineExceeded),
```

Comment:
@nojnhuh this is the only way I could get this test to pass, I tried using:

`azure.CreateOrUpdateAsyncOpts{ResumeToken: "", Parameters: gomock.Any()}`

as the 3rd argument to `c.CreateOrUpdateAsync` and it was complaining:

```
            Got: { {<nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> map[] <nil> <nil> <nil>}} (azure.CreateOrUpdateAsyncOpts)
            Want: is equal to { is anything} (azure.CreateOrUpdateAsyncOpts)
```

Using `gomock.Any()` to represent the now-current `azure.CreateOrUpdateAsyncOpts` type seems like cheating...